listenbrainz/webserver/views/stats_api.py

```diff
@@ -17,55 +17,59 @@ def get_artist(user_name):
 
         {
             "payload": {
-                "all_time": {
-                    "artists": [
-                        {
-                           "artist_mbids": [93e6118e-7fa8-49f6-9e02-699a1ebce105],
-                           "artist_msid": "d340853d-7408-4a0d-89c2-6ff13e568815",
-                           "artist_name": "The Local train",
-                           "listen_count": 385
-                        },
-                        {
-                           "artist_mbids": [ae9ed5e2-4caf-4b3d-9cb3-2ad626b91714],
-                           "artist_msid": "ba64b195-01dd-4613-9534-bb87dc44cffb",
-                           "artist_name": "Lenka",
-                           "listen_count": 333
-                        },
-                        {
-                           "artist_mbids": [cc197bad-dc9c-440d-a5b5-d52ba2e14234],
-                           "artist_msid": "6599e41e-390c-4855-a2ac-68ee798538b4",
-                           "artist_name": "Coldplay",
-                           "listen_count": 321
-                        }
-                    ],
-                    "count": 3
-                },
-            },
-
-           "last_updated": 1588494361,
-           "user_id": "John Doe"
+                "artists": [
+                    {
+                       "artist_mbids": [93e6118e-7fa8-49f6-9e02-699a1ebce105],
+                       "artist_msid": "d340853d-7408-4a0d-89c2-6ff13e568815",
+                       "artist_name": "The Local train",
+                       "listen_count": 385
+                    },
+                    {
+                       "artist_mbids": [ae9ed5e2-4caf-4b3d-9cb3-2ad626b91714],
+                       "artist_msid": "ba64b195-01dd-4613-9534-bb87dc44cffb",
+                       "artist_name": "Lenka",
+                       "listen_count": 333
+                    },
+                    {
+                       "artist_mbids": [cc197bad-dc9c-440d-a5b5-d52ba2e14234],
+                       "artist_msid": "6599e41e-390c-4855-a2ac-68ee798538b4",
+                       "artist_name": "Coldplay",
+                       "listen_count": 321
+                    }
+                ],
+                "count": 3,
+                "range": "all_time"
+                "last_updated": 1588494361,
+                "user_id": "John Doe"
             }
         }
 
     .. note::
-       - ``artist_mbids`` and ``artist_msid`` are optional fields and may not be present in all the responses
-       - As of now we are only calculating ``all_time`` statistics for artist.
-         However, we plan to add other time intervals in the future.
+        - This endpoint is currently in beta
+        - ``artist_mbids`` and ``artist_msid`` are optional fields and may not be present in all the responses
+        - As of now we are only calculating ``all_time`` statistics for artist.
+          However, we plan to add other time intervals in the future.
```

Comment:
If different ranges are going to be supported, would it not make sense not to use named ones in the response, but instead use something like `timespan: { from: <unix_ts>, to: <unix_ts> }`?
The current always-all-time-stats could set those either to the timestamps of the user's oldest and newest listen (if readily available), or even just the unix epoch as `from` and the current timestamp as `to`. That way, the payload is stable and usable, regardless of what kinds of ranges get supported later.

Reply:
If the user's oldest and newest listen timestamps are known, it would especially useful to have them, because you could show "Top <count> Artists for <user> (over <textual description of timespan>)", e.g "Top 10 Artists for Zastai (over 10 years and 5 months)".

"All time" is not very useful, because it could cover anything from "5 minutes" (if the user only has one submitted listen) to "18 years" (if the user has been using last.fm since it started).

Reply:
We currently calculate user statistics in a batch every 15 days so answering queries for arbitrary time periods is not possible. Can you open a ticket for the functionality so that we can look into it in the future. Thanks.